backend/app/api/routes/documents.py

```diff
@@ -290,6 +311,23 @@ async def process_multiple_documents(
     return {"message": "Processing started for multiple files", "documents": results}
 
 
+@router.get("/by-course/{course_id}")
+def list_documents_by_course(session: SessionDep, current_user: CurrentUser, course_id: uuid.UUID) -> Any:
+    """List documents for a course with basic fields."""
+    docs = session.exec(select(Document).where(Document.course_id == course_id)).all()
+    return [
```

Comment:
Return the docs and include the response_model. The rest will be handled for you by FastAPI and Pydantic.